### PR TITLE
[cxx-interop] Add `-cxx-interoperability-mode` driver flag; deprecate `-…

### PR DESCRIPTION
**Pull request by zoecarver**
…enable-experimental-cxx-interop`.

Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves #NNNNN, fix apple/llvm-project#MMMMM.

### CODE REVIEW

**Comment by zoecarver** [comment]
- [ ] Rebase
- [ ] Update existing tests
- [ ] Add tests for new flag and diagnostics 
- [ ] Update swift driver
- [ ] Update docs for corresponding swiftpm updates

File: lib/Frontend/CompilerInvocation.cpp [hunk 1]
```diff
@@ -429,6 +429,20 @@ static void SaveModuleInterfaceArgs(ModuleInterfaceOptions &Opts,
...
+enum class CxxCompatMode {
+  invalid,
+  enabled,
+  off
+};
+
+static CxxCompatMode validateCxxInteropCompatibilityMode(StringRef mode) {
+  if (mode == "off")
+    return CxxCompatMode::off;
+  if (mode == "swift-5.7")
```
Comment (zoecarver) [inline lib/Frontend/CompilerInvocation.cpp:441]:
```suggestion
  if (mode == "swift-5.9")
```

And similar

**Comment by zoecarver** [comment]
@swift-ci please smoke test

**Comment by zoecarver** [comment]
Once https://github.com/apple/swift-driver/pull/1306 lands I'll make another patch to update the tests in this repo.

File: include/swift/Option/Options.td [hunk 2]
```diff
@@ -663,9 +663,15 @@ def enable_experimental_concise_pound_file : Flag<["-"],
...
 def enable_experimental_cxx_interop :
   Flag<["-"], "enable-experimental-cxx-interop">,
-  Flags<[FrontendOption, HelpHidden, ModuleInterfaceOption]>,
+  Flags<[NoDriverOption, FrontendOption, HelpHidden, ModuleInterfaceOption]>,
   HelpText<"Enable experimental C++ interop code generation and config directives">;
 
+def cxx_interoperability_mode :
+  Joined<["-"], "cxx-interoperability-mode=">,
+  Flags<[FrontendOption, ModuleInterfaceOption]>,
+  HelpText<"Enables C++ interoperability; requires compatbility version to be"
```
Comment (artemcm) [inline include/swift/Option/Options.td:672]:
```suggestion
  HelpText<"Enables C++ interoperability; requires compatbility version to be "
```

**Comment by zoecarver** [comment]
@swift-ci please smoke test

File: docs/CppInteroperability/GettingStartedWithC++Interop.md [hunk 3]
```diff
@@ -32,15 +32,9 @@ module CxxTest {
 Add the C++ module to the include path and enable C++ interop:
 - Navigate to your project directory
 - In `Project` navigate to `Build Settings` -> `Swift Compiler`
-- Under `Custom Flags` -> `Other Swift Flags` add `-Xfrontend -enable-experimental-cxx-interop`
+- Under `Custom Flags` -> `Other Swift Flags` add `-cxx-compatibility-mode=swift-5.9`
```
Comment (zoecarver) [inline docs/CppInteroperability/GettingStartedWithC++Interop.md:35]:
```suggestion
- Under `Custom Flags` -> `Other Swift Flags` add `-cxx-interoperability-mode=swift-5.9`
```

Looks like this one got lost in the rebase.. oops

**Pull request merged**